server/neptune/workflows/internal/pr/revision/policy/handler.go

```diff
@@ -72,39 +79,47 @@ func (f *FailedPolicyHandler) Handle(ctx workflow.Context, revision revision.Rev
 		if reviewRequest.Revision != revision.Revision {
 			action = onSkip
 		}
-		scope.SubScopeWithTags(map[string]string{metricNames.SignalNameTag: "pr-approval"}).
+		scope.SubScopeWithTags(map[string]string{metricNames.SignalNameTag: "pr-review"}).
 			Counter(metricNames.SignalReceive).
 			Inc(1)
 	})
 	onTimeout := func(f workflow.Future) {
 		_ = f.Get(ctx, nil)
 		action = onPollTick
-		scope.SubScopeWithTags(map[string]string{metricNames.PollNameTag: "pr-approval"}).
+		scope.SubScopeWithTags(map[string]string{metricNames.PollNameTag: "pr-review"}).
 			Counter(metricNames.PollTick).
 			Inc(1)
 	}
-	cancelTimer, _ := s.AddTimeout(ctx, 5*time.Minute, onTimeout)
+	cancelTimer, _ := s.AddTimeout(ctx, 10*time.Minute, onTimeout)
 
 	for {
-		if len(failedPolicies) == 0 {
+		if len(failingTerraformWorkflows) == 0 {
 			break
 		}
 		s.Select(ctx)
 		switch action {
 		case onSkip:
 			continue
-		case onApprovalSignal:
-			failedPolicies = f.handle(ctx, revision, failedPolicies)
 		case onPollTick:
 			// TODO: evaluate a better polling rate for approvals, or remove all together
 			cancelTimer()
 			cancelTimer, _ = s.AddTimeout(ctx, 10*time.Minute, onTimeout)
-			failedPolicies = f.handle(ctx, revision, failedPolicies)
 		}
+
+		// onPollTick and onReviewSignal actions, filter out failing policies that have been approved and identify if
+		// any previously failing terraform workflows are now successful
+		remainingFailedPolicies := f.filterOutBypassedPolicies(ctx, revision, failingTerraformWorkflows)
+		successfulTerraformWorkflows := partitionWorkflowsByResult(failingTerraformWorkflows, remainingFailedPolicies, true)
+		failingTerraformWorkflows = partitionWorkflowsByResult(failingTerraformWorkflows, remainingFailedPolicies, false)
```

Comment:
Doesnt seem like we use `failingTerraformWorkflows` past this point? Or is this a pointer to something used elsewhere?

Reply:
Used at the start of each for loop to check if we should continue iterating:
https://github.com/lyft/atlantis/pull/693/files#diff-da36521dd456bb50ddfca426bd4c437ebd9383d39dee9634bc006ceead746902R95-R102